File: OpenTokFs.RequestTypes/OpenTokRtmpDestination.cs
using Newtonsoft.Json;

namespace OpenTokFs.RequestTypes {
    public class OpenTokRtmpDestination {
        [JsonProperty("id", NullValueHandling = NullValueHandling.Ignore)]
        public string Id { get; set; }

        [JsonProperty("serverUrl")]
        public string ServerUrl { get; set; }

        [JsonProperty("streamName")]
        public string StreamName { get; set; }

        public override string ToString() => $"{ServerUrl}/{StreamName} ({Id ?? "no ID"})";
    }
}

File: OpenTokFs/RequestTypes/OpenTokRtmpDestination.fs
namespace OpenTokFs.RequestTypes

type OpenTokRtmpDestination() =
    member val id: string = null with get, set
    member val serverUrl = "" with get, set
    member val streamName = "" with get, set
    override this.ToString () = String.concat " " [
        $"{this.serverUrl}/{this.streamName}"
        if not (isNull this.id) then $"({this.id})"
    ]
